src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionDuplicatedArgsInspection.kt

```diff
@@ -31,18 +31,24 @@ class SmkSectionDuplicatedArgsInspection : SnakemakeInspection() {
         ) {
             val args = argumentList?.arguments ?: emptyArray()
             if (args.size > 1) {
-                val setOfDeclaredArguments = mutableSetOf<PsiElement>()
+                val setOfDeclaredArguments = mutableSetOf<String>()
 
                 args.forEach { arg ->
-                    if (setOfDeclaredArguments.any{x -> arg.textMatches(x)}) {
-                        registerProblem(
-                                arg,
-                                SnakemakeBundle.message("INSP.NAME.section.duplicated.args.message",
-                                        section.sectionKeyword!!),
-                                RemoveArgumentQuickFix()
-                        )
-                    } else {
-                        setOfDeclaredArguments.add(arg)
+
+                    /* PyKeywordArgumentImpl is checked by SmkSyntaxErrorAnnotator */
+                    if (arg !is PyKeywordArgumentImpl) {
```

Comment:
Please use PSI open API (i.e. interfaces) where possible and avoid using the interfaces implementation classes (i.e. use `PyKeywordArgument` instead of `PyKeywordArgumentImpl`). In future, the exact impl classes could be renamed or alternative implementations could be added, interface API is more stable. We don't need here any specific methods of `PyKeywordArgumentImpl`

Reply:
Done cb7df88